src/Entity/Storage/AppStorage.php

```diff
@@ -94,6 +94,31 @@ public function loadUnchanged($id) {
     return parent::loadUnchanged($id);
   }
 
+  /**
+   * Load app by UUID.
+   *
+   * This function is more efficient than loadUnchanged(),
+   * because it does not need to cover the case
+   * when loading is done by App name.
+   *
+   * @param string $uuid
+   *   App UUID.
+   *
+   * @return \Drupal\apigee_edge\Entity\AppInterface|null
+   *   The unchanged entity, or NULL if the entity cannot be loaded.
+   *
+   * @TODO: this method should be also available in the AppStorageInterface,
+   *   but that would be a breaking change,
+   *   so we can only add that in the next major version of the module.
+   */
+  public function loadUnchangedById(string $uuid): ?AppInterface {
```

Comment:
If everything is "UUID" now, should not this be also `loadUnchangedByUuid`?

Reply:
I am not sure it should be, it would definitely be inconsistent with rest of the code. This module uses id instead of uuid all over the place.

This line: https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/src/ParamConverter/DeveloperAppNameConverter.php#L90

would have to become:            
$entity = $app_storage->loadUnchangedByUuid($app_id);

The module has variables like $app_id, when in fact, according to this thinking, it should be $app_uuid. Similarly, you have $developer_id, etc

https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/src/ParamConverter/DeveloperAppNameConverter.php#L78

Reply:
@en-vcracan I see what you mean. Though having a `loadUnchanged($id)` and a `loadUnchangedById` which accepts a `uuid` can be misleading.

Reply:
@arshad we can go back and rename $uuid to $id, that's no problem. I myself think that would be best. But if you confirm you want this to be uuid, then I will change the function name. Let me know.

Reply:
@arshad we can go back and rename $uuid to $id, that's no problem. I myself think that would be best. But if you confirm you want this to be uuid, then I will change the function name. Let me know.

Reply:
@en-vcracan Yes. Let's rename it to `loadUnchangedByUuid`. Thank you.

Reply:
Renamed